src/bp/ui-admin/src/reducers/versions.ts

```diff
@@ -0,0 +1,82 @@
+import axios from 'axios'
+import _ from 'lodash'
+import moment from 'moment'
+
+interface GithubRelease {
+  version: string
+  details: string
+  githubUrl: string
+  releaseDate: Date
+  daysAgo: string
+}
+
+export interface VersionState {
+  currentVersion: string
+  latestReleases: GithubRelease[]
+}
+
+const RECEIVE_CURRENT_VERSON = 'version/RECEIVE_CURRENT_VERSON'
+const RECEIVE_LATEST_RELEASES = 'version/RECEIVE_LATEST_RELEASES'
+
+export const fetchCurrentVersion = () => {
+  return async (dispatch, getState) => {
+    if (getState().version.currentVersion) {
+      return
+    }
```

Comment:
I don't think reducers should have that kind of logic. If I call fetchCurrentVersion, i expect the call to be made, acting as a "dumb method". The component should decide if it needs to fetch the new version (like we do everywhere else)

Reply:
I'm pretty sure not. reducer is not dumb this is where all the logic reside. And moving this logic here keeps defined check away from the rendering.

Reply:
I agree that they could keep some of that logic, but everywhere else we do that check in useEffect, so i'd leave it that way so actions "fetch*" are consistent across the admin ui.

Reply:
as you wish. done.